docs/articles/documentation/extending-testcafe/custom-browser-provider-plugin/index.md

```diff
@@ -0,0 +1,200 @@
+---
+layout: docs
+title: Custom Browser Provider Plugin
+permalink: /documentation/extending-testcafe/custom-browser-provider-plugin/
+---
+# Custom Browser Provider Plugin
+
+TestCafe has built-in support for testing in a number of popular browsers (see [Using Browsers](../../using-testcafe/common-concepts/using-browsers.md)).
+If you need to use a different browser (for example, a browser from a cloud service), you can create a **custom browser provider plugin**.
+The provider should expose methods for performing common actions on the browser windows during testing: opening and closing it, taking screenshot, etc.
+
+To create a browser provider plugin, go through the following steps.
+
+* [Generating Browser Provider Project](#generating-browser-provider-project)
+* [Implementing the Browser Provider](#implementing-the-browser-provider)
+* [Building the Provider](#building-the-provider)
+* [Using the Provider Development Version](#using-the-provider-development-version)
+* [Publishing the Provider to NPM](#publishing-the-provider-to-npm)
+* [Reference](#reference)
+
+> To run tests in [Sause Labs](https://saucelabs.com) or [PhantomJS](http://phantomjs.org/) browsers, you can use the existing [testcafe-browser-provider-saucelabs](https://github.com/DevExpress/testcafe-browser-provider-saucelabs/) or [testcafe-browser-provider-phantomjs](https://github.com/DevExpress/testcafe-browser-provider-phantomjs) plugin.
+
+## Generating Browser Provider Project
+
+To create a browser provider project, use the [TestCafe browser provider generator](https://github.com/DevExpress/generator-testcafe-browser-provider).
+The generator will scaffold out the plugin, so that you only need to write a few lines of code.
+
+First, install [Yeoman](http://yeoman.io) and `generator-testcafe-browser-provider` using [npm](https://www.npmjs.com/).
+
+```bash
+npm install -g yo
+npm install -g generator-testcafe-browser-provider
+```
+
+Create a new directory where the generator should place your scaffolded project files and go into it.
+
+```bash
+mkdir my-provider
+cd my-provider
+```
+
+> It is recommended to name the directory as you would like to name the browser provider project. When you run the generator, it will automatically suggest the browser provider name that matches the browser provider directory name.
+>
+> The generator will also automatically create the provider package name that consists of two parts - the `testcafe-browser-provider-` prefix and the name of the provider itself, for example, `testcafe-browser-provider-my-provider`.
+>
+> **Important:** If you name the provider package manually, it is required that its name starts with the `testcafe-browser-provider-` prefix. Otherwise, TestCafe will be unable to recognize the plugin.
+
+Then run the browser provider generator to create a new project:
+
+```bash
+yo testcafe-browser-provider
+```
+
+The generator will ask you for information about the browser provider.
```

Comment:
same as in https://github.com/DevExpress/testcafe/pull/725/files#r76282948
sorry)